tapTrainer.m

```diff
@@ -95,6 +98,9 @@
         taps = []; 
         istap = false;
 
+        % counter for audio-track index (used only for tracks, but we can
+        % pass it to makeStimTrain all the time)
+        soundIdx = 0; 
         
```

Comment:
I just run the tapTraining. The first phase/level changes with the entertaining bugz-4_hugz! very cool. Unfortunately, I did not follow exactly what you are doing in `makeStimTrain`. In the first level, you are changing couple of audio tracks? I thought that was 1 audio.

Reply:
In `makeStimTrain`, I just look what sort of stimulus I need to synthesize: 

1. If it's an **audio track**, it has already been preloaded in `getTrainingParams`, but I need to get a short segment (equivalent to window duration), and mix it with the cue at an appropriate dB level. This is a little bit different from having a repeating pattern that I can just synthesize and return, because I need to know where exactly in the long audio track we are atm. That's why I added the parameter `soundIdx`, which keeps the index of the last sample  from the audio track that has been pushed to the sound buffer. So during the training, I keep calling `makeStimTrain`, and getting small successive chunks of the audio track, mixed with cue with the dB I ask for and pushing them into the sound buffer. 

2. If it's a **pattern**, I need to synthesize it etc., just like we did in the previous versions.